FinSearchUnified/Tests/Subscriber/WidgetsTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace FinSearchUnified\Tests\Subscriber;
+
+use Enlight_Controller_Request_RequestHttp;
+use Enlight_Controller_Response_ResponseHttp;
+use Enlight_Hook_HookArgs;
+use ReflectionException;
+use Shopware\Components\Test\Plugin\TestCase;
+use Shopware_Controllers_Widgets_Listing;
+
+class WidgetsTest extends TestCase
+{
+    public function setUp()
+    {
+        parent::setUp();
+
+        // reset global variable as it conflicts with existing tests in this class
+        $_GET = [];
+    }
+
+    /**
+     * @dataProvider searchParameterProvider
+     *
+     * @param array $requestParameters
+     * @param string|null $expectedSearchParameter
+     * @param string $expectedMessage
+     *
+     * @throws ReflectionException
+     */
+    public function testBeforeListingCountAction(
+        array $requestParameters,
+        $expectedSearchParameter,
+        $expectedMessage
+    ) {
+        $request = new Enlight_Controller_Request_RequestHttp();
+        $request->setControllerName('listing')
+            ->setActionName('listingCount')
+            ->setModuleName('widgets')
+            ->setParams($requestParameters);
+
+        /** @var Shopware_Controllers_Widgets_Listing $subject */
+        $subject = Shopware_Controllers_Widgets_Listing::Instance(
+            Shopware_Controllers_Widgets_Listing::class,
+            [
+                $request,
+                new Enlight_Controller_Response_ResponseHttp()
+            ]
+        );
+
+        // Create mocked args for getting Subject and Request
+        $args = $this->getMockBuilder(Enlight_Hook_HookArgs::class)
```

Comment:
It should be possible to use the class directly instead of a mock.

Reply:
I remember why I mocked it, due to backward compatibility as the test were failing for older shopware versions